lib/chef/resource/yum_package.rb

```diff
@@ -38,7 +38,7 @@ def arch(arg=nil)
         set_or_return(
           :arch,
           arg,
-          :kind_of => [ String ]
+          :kind_of => [ String, Array ]
```

Comment:
The first syntax is what I was fixing. The second syntax won't work because arch isn't an option on package, but it will work assuming you do `yum_package`